src/User/Controller/SettingsController.php

```diff
@@ -347,9 +354,43 @@ public function actionNetworks()
         );
     }
 
+    public function actionConsent()
+    {
+        $user = Yii::$app->user->identity;
+
+        if (!!$user->gdpr_consent)
+            $this->redirect(['profile']);
+
+        $model = new DynamicModel(['gdpr_consent']);
+        $model->addRule('gdpr_consent', 'boolean');
+        $model->addRule('gdpr_consent', 'default', ['value' => 0, 'skipOnEmpty' => false]);
+        $model->addRule('gdpr_consent', 'compare', [
+            'compareValue' => true,
+            'message' => Yii::t('usuario', 'Your consent is required to work with this site'),
+            'when' => function () {
+                return $this->module->enableGDPRcompliance;
+            }
+        ]);
+
+        if ($model->load(Yii::$app->request->post()) && $model->validate()) {
+            $user->updateAttributes([
```

Comment:
Maybe a method within the model would make more sense in order to keep the action cleaner. So, could be: 

```
if ($model->load(Yii::$app->request->post()) && $model->validate() && $user->consentGdpr()) {
   return $this->redirect(['profile']); 
}
```